src/stratis_cli/_actions/_logical.py

```diff
@@ -138,12 +162,26 @@ def destroy_volumes(namespace):
             .require_unique_match(True)
             .search(managed_objects)
         )
+
+        requested_names = frozenset(namespace.fs_name)
+
+        pool_filesystems = {
+            MOFilesystem(info).Name(): op
+            for (op, info) in filesystems(props={"Pool": pool_object_path}).search(
+                managed_objects
+            )
+        }
+        already_removed = requested_names.difference(frozenset(pool_filesystems.keys()))
+
+        if already_removed != frozenset():
+            raise StratisPartialChangeError(
+                "destroy",
+                requested_names.difference(already_removed),
+                list(already_removed),
+            )
+
         fs_object_paths = [
```

Comment:
This make sense so we don't do unnecessary work if we already know it will fail.